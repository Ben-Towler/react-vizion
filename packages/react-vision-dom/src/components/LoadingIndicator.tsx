import React from 'react';
import classnames from 'classnames';

type Props = {
  isLoading?: boolean;
};

const LoadingIndicator: React.FunctionComponent<Props> = ({ isLoading }) => {
  return (
    <div>
      <span className={classnames({ loading: isLoading })} />
    </div>
  );
};

export default LoadingIndicator;
