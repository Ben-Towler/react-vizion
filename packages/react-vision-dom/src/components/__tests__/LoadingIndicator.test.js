import React from 'react';
import renderer from 'react-test-renderer';
import Adapter from 'enzyme-adapter-react-16';
import Enzyme, { shallow, mount } from 'enzyme';
import LoadingIndicator from '../LoadingIndicator';

Enzyme.configure({ adapter: new Adapter() });

describe('Loading Indicator', () => {
  it('applies its default props', () => {
    const instance = renderer.create(<LoadingIndicator refine={() => null} />);

    expect(instance.toJSON()).toMatchSnapshot();

    instance.unmount();
  });

  it('should render with .loading class when isLoading prop is true', () => {
    const wrapper = mount(<LoadingIndicator isLoading={true} />);
    expect(wrapper.exists('.loading')).toBeTruthy();
  });

  it('should not render with .loading class when isLoading prop is false', () => {
    const wrapper = mount(<LoadingIndicator isLoading={false} />);
    expect(wrapper.exists('.loading')).toBeFalsy();
  });
});
