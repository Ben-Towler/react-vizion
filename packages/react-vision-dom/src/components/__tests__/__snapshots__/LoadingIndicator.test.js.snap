// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loading Indicator applies its default props 1`] = `
<div>
  <span
    className=""
  />
</div>
`;
